Foundation/URLSession/URLSessionTask.swift

```diff
@@ -552,24 +552,30 @@ extension _ProtocolClient : URLProtocolClient {
             session.delegateQueue.addOperation {
                 delegate.urlSession(session, task: task, didCompleteWithError: nil)
                 task.state = .completed
-                task.workQueue.async {
+                session.workQueue.async {
                     session.taskRegistry.remove(task)
                 }
             }
         case .noDelegate:
             task.state = .completed
-            session.taskRegistry.remove(task)
+            session.workQueue.async {
```

Comment:
Sorry for necroposting but this and this https://github.com/apple/swift-corelibs-foundation/pull/1625/files#diff-14c6cb724eead661596e4468ae5ca638R625 looks like overkill
There is no thread switching with `session.delegateQueue.addOperation`
And if this method itself not synchronized race will be earlier at line 545